build/Makefile

```
@@ -101,6 +106,13 @@ ifdef DOCKER_RUN
 	ensure-build-image += ensure-build-image
 endif
 
+ifdef INCREMENTAL_BUILD
+	# keep a cache of files built by Go across docker invocations in a local directory.
+	common_mounts += -v $(CURDIR)/.gocache:/root/.cache/go-build
```

Rather than use CURDIR, which is the working directory, let's explicitly put it in the $(build_path), then it never moves, regardless of where people call the Makefile from.

```suggestion
 	common_mounts += -v $(build_path)/.gocache:/root/.cache/go-build
```

Then we can also add a `mkdir -p` to the `ensure-build-config` target, so that the directory always is created before being mounted (I thought it should throw an error if it's not there - has that stopped happening?)

---

Docker appears to be creating the directory automatically.